airbyte-server/src/main/java/io/airbyte/server/handlers/SourceHandler.java

```diff
@@ -295,14 +301,14 @@ private void persistSourceConnection(final String name,
                                        final JsonNode configurationJson,
                                        final ConnectorSpecification spec)
       throws JsonValidationException, IOException {
+    final JsonNode oAuthMaskedConfigurationJson = oAuthConfigSupplier.maskSourceOAuthParameters(sourceDefinitionId, workspaceId, configurationJson);
```

Comment:
New: Source and Destination update APIs now mask global oAuth parameters if they're present. For OSS users without any oauth parameters in their db this will have no effect.

These endpoints were already handling splitting out the secrets to store separately from the config - we're just adding to this and _also_ masking global oauth params.

Reply:
for my understanding, how is this related to persisting configs?

Reply:
This is necessary because when we receive a config control message from a connector, it will include those global oauth parameters as part of the config (e.g. our oauth app's client_id + secret). These oauth params are not provided by the user, and are instead injected by the platform. We don't want to persist those as part of the user's secrets when writing back the updated configuration, so we need to mask them again.

Reply:
Shall we mask these global oAuth parameters on the connector side in the CDK?

Reply:
No, the connector doesn't know what's a global auth param and what's provided directly by the user. Note that in the case of OSS users can provide their own oauth client ids and secrets as part of the config, and global auth params are really only enabled in Cloud. I think considering that the control message contains secrets and handling that on the platform side makes the interface easier to work with, and less to have connector devs worry about.